services/QuillLMS/spec/controllers/api/v1/activity_sessions_controller_spec.rb

```diff
@@ -39,6 +41,27 @@
       end
     end
 
+    context 'user_activity_classification counts increment when they should' do
+      it 'should create a new user_activity_classification row if necessary' do
+        records = UserActivityClassification.count
+        put :update, params: { id: activity_session.uid, state: 'finished' }, as: :json
+        expect(UserActivityClassification.count).to eq(records + 1)
+      end
+
+      it 'should increment an existing user_activity_classification row if one exists' do
+        start_count = 10
+        UserActivityClassification.create(user: user, activity_classification: activity_classification, count: start_count)
+        put :update, params: { id: activity_session.uid, state: 'finished' }, as: :json
+        expect(UserActivityClassification.find_by(user: user, activity_classification: activity_classification).count).to eq(start_count + 1)
+      end
+
+      it 'should not touch user_activity_classification counts if the state of the session is not "finished"' do
+        records = UserActivityClassification.count
+        put :update, params: { id: activity_session.uid }, as: :json
+        expect(UserActivityClassification.count).to eq(records)
+      end
+    end
+
```

Comment:
rspec has a [built-in change matcher](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers) that you can use to track these types of specs.  Basically, it's expect { ..some block of code that changes object } .to change { object.method }.by(val).  Initially, I found this pattern hard to grok but I now prefer it gives a nice scaffold for this types of tests, especially when ones involving counts on database objects.

A rewrite would look something like this:

```
      it 'should create a new user_activity_classification row if necessary' do
        expect {
          put :update, params: { id: activity_session.uid, state: 'finished' }, as: :json
        }.to change { UserActivityClassification.count }.by(1)
      end

      it 'should increment an existing user_activity_classification row if one exists' do
        start_count = 10
        uac = UserActivityClassification.create(user: user, activity_classification: activity_classification, count: start_count)

        expect {
          put :update, params: { id: activity_session.uid, state: 'finished' }, as: :json
          uac.reload
        }.to change { uac.count }.from(start_count).to(start_count + 1)
      end

      it 'should not touch user_activity_classification counts if the state of the session is not "finished"' do
        expect {
          put :update, params: { id: activity_session.uid }, as: :json
        }.to_not change { UserActivityClassification.count }
      end
```

There's also an alternative parameterized form for `change` where instead of a block you pass parameters change(UserActivityClassification, :count).

Reply:
Seems reasonable.  Updated.

Reply:
@anathomical I just realized that rubocop doesn't like the block form of change.  (I hit the same issue [here](https://app.circleci.com/pipelines/github/empirical-org/Empirical-Core/10239/workflows/d8523f85-b444-4863-a45f-c581ffa32ee0/jobs/174896))  Good news though is the parameterized version should work fine:

```
change (UserActivityClassification, :count)
change (uac, :count)
change (UserActivityClassification.count)
```